## Lab  8.4 | Inferential statistics - ANOVA

In [1]:
import statsmodels.api as sm
from statsmodels.formula.api import ols
import pandas as pd

### Context

Suppose you are working as an analyst in a microprocessor chip manufacturing plant. You have been given the task of analyzing a plasma etching process with respect to changing Power (in Watts) of the plasma beam. Data was collected and provided to you to conduct statistical analysis and check if changing the power of the plasma beam has any effect on the etching rate by the machine. You will conduct ANOVA and check if there is any difference in the mean etching rate for different levels of power. You can find the data `anova_lab_data.xlsx` file in the `lab-8.4` folder  

- State the null hypothesis
- State the alternate hypothesis
- What is the significance level
- What are the degrees of freedom of model, error terms, and total DoF

Data was collected randomly and provided to you in the table as shown: [link to the image - Data](https://education-team-2020.s3-eu-west-1.amazonaws.com/data-analytics/7.05/7.05-lab_data.png)


![alt text](lab_data.png)

![alt text](f_table_lab.png)

1. Here are the steps that you would need to work on:

- Null hypothesis

H0:There is no significant difference in the mean etching rates for different levels of power.


- Alternate hypothesis

H1: there is at least one significant difference in the mean etching rates for different levels of power.

- Level of significance

The significance level (α) is not explicitly stated in the problem, so we will assume a common significance level of 0.05.

- DoF

As we see on the table
- Model between groups : 2
- Error within groups: 12
- Total: 14

- Test statistic

2. In this section, use Python to conduct ANOVA.
What conclusions can you draw from the experiment and why?

In [1]:
import statsmodels.api as sm
from statsmodels.formula.api import ols
import pandas as pd

In [2]:
data = pd.read_excel('anova_lab_data.xlsx', sheet_name='data_collected')
data

Power   Etching Rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

In [ ]:
#Renaming variables so it doesn't raise error:

In [4]:
data.rename(columns={'Etching Rate':'Etching_Rate'}, inplace=True)

In [5]:
data.rename(columns={'Power ':'Power'}, inplace=True)

In [6]:
model = ols('Etching_Rate ~ C(Power)',data=data).fit()
table = sm.stats.anova_lm(model)
print(table)

            df     sum_sq   mean_sq          F    PR(>F)
C(Power)   2.0  18.176653  9.088327  36.878955  0.000008
Residual  12.0   2.957240  0.246437        NaN       NaN


As the F statistic is 36.878955 and a p-value of <0.0008, which is much smaller than the significance level of 0.05, we reject the null hypothesis.

There is sufficient evidence to say that changing the power of the plasma beam has a significant effect on the etching rate.